### ISSUE TITLE
Invalid customer ID while trying to get my customerClient list

### ISSUE THREAD

**Issue by delentej** [Opteo/google-ads-api#302]
Hello, I'm upgrading my code, using the google-ads-api from npm on version 5.2.0, and I'm trying to get all my customer from my manager account. I'm getting the following error code when i am using the report() function : 

    GoogleAdsFailure {
      errors:
       [ GoogleAdsError {
           error_code: [ErrorCode],
           message: 'Invalid customer ID \'\'.' } ],
      request_id: 'v5YJ5e3CMzBPsDjq9nQZuA'
     }    

This is probably coming form the attributes field that should be wrong. I am trying to get at least the descriptive name and the id of the customer.

This is my code :

    const client = new googleApi.GoogleAdsApi({
        client_id: clientId,
        client_secret: clientSecret,
        developer_token: developperToken,
    });

    const manager = client.Customer({ 
        customer_account_id: managerAccountId,
        refresh_token: refreshToken,
        login_customer_id: managerAccountId
    });

    let customerClients;
    try {
        customerClients = await manager.report({
            entity: 'customer_client',
            attributes: ['customer_client.id','customer_client.descriptive_name']
        });
    } catch (error) {
        console.log(error);
    }

**Comment by delentej** [comment]
also i'm having the same issue with campaign. I used to get all my campaign with the following code : 

            const accountCampaigns = await account.report({
                entity: 'campaign',
                attributes: ['campaign.id', 'campaign.name', 'campaign.status', 'campaign.start_date', 'campaign.end_date']
            });

I'm having the same issue.

**Comment by WillCooter** [comment]
This appears to be a duplicate of #279. Are you certain that you are following [Google's API Call Structure](https://developers.google.com/google-ads/api/docs/concepts/call-structure?hl=en#request_headers) correctly? For example, have you removed any hyphens (`-`) from your login customer id?

**Comment by LuKaa** [comment]
@delentej I had the same error changing from v4 to v5.2
Try changing **customer_account_id** to **customer_id** in the customer object and make sure your remove all hyphens (`-`) from the id.

**Comment by WillCooter** [comment]
Thanks for you help @LuKaa. Did this solve your problem @delentej?

**Comment by MitulGedeeya** [comment]
@WillCooter I started getting this error while using Report API.

**await customer.report()**

getting following message for all ad accounts.

```
GoogleAdsFailure {
  errors: [
    GoogleAdsError {
      error_code: [ErrorCode],
      message: "Invalid customer ID ''."
    }
  ],
  request_id: '5qJwTFD3XIuWfJKCXNfonw'
}

```

**Comment by lushu** [comment]
Hi @WillCooter, I have got this 'Invalid customer ID' error when I tried to move to new v8( mainly reporting API). Here are my fundings:

1.  Run exactly same tests for different `google-ads-api` libs `5.2.0` and `8.0.2`:
    -  `5.2.0` all worked but `8.0.2` all failed except one function.
    - The only worked function in `8.0.2` is `customer.googleAdsFields.searchGoogleAdsFields` which is actually doesn't care the customer ID that you put in.
2. I also tried the some same tests for `google-ads-node` libs `v5.0.0` vs `6.1.2`:
    - `5.0.0` worked but `6.1.2` failed.
    - I tried to change the `customer_id` to `customerId` where `SearchGoogleAdsRequest` was initialized -- it didn't make too much sense based on the [code](https://github.com/Opteo/google-ads-node/blob/b375fda83c4346d44858c3f4b2229c340d67d74a/package/googleads-nodejs/protos/google/ads/googleads/v8/services/google_ads_service.proto#L384) but it worked.

Besides that,I also noticed the response of `searchGoogleAdsFields` changed properties' names from snake to camel. So did the response of `google-ads-node` of `6.1.2`.

Is this a expected behavior of the lib `google-ads-node`? Is there any settings to change it back to 'snake' naming? Thanks in advance!

**Comment by lushu** [comment]
Hi @WillCooter , do you have any estimated time for this? The v7 is going to be deprecated in the next week. Thank you.

**Comment by WillCooter** [comment]
@delentej @LuKaa @lushu @aohren This should be fixed in the latest version of the library (`v8.1.0`)
